### PR TITLE
Fixes for #196 & #372

### PR DESCRIPTION
**Pull request by karthik-sethuraman**
#196: Partial fix - Added RPC augments for photonic Media
#372 : Added key to complex data-types used as list, added key to list parameters of RPCs

### CODE REVIEW

**Comment by Andrea-Campanella** [comment]
+1 for me. Can you please share on ODTN mailing list to check for comments ?

File: YANG/tapi-photonic-media@2018-10-16.tree [hunk 1]
```diff
@@ -183,3 +183,270 @@ module: tapi-photonic-media
              +--ro power-spectral-density?   decimal64
   augment /tapi-common:context/tapi-topology:topology-context/tapi-topology:topology/tapi-topology:node/tapi-topology:owned-node-edge-point/tapi-connectivity:cep-list/tapi-connectivity:connection-end-point:
     +--ro media-channel-assembly-spec
+  augment /tapi-common:get-service-interface-point-list/tapi-common:output/tapi-common:sip:
```
Comment (arthurMll) [inline YANG/tapi-photonic-media@2018-10-16.tree]: I think the following is missing:
augment /tapi-common:context/tapi-common:service-interface-point:
    +--rw media-channel-service-interface-point-spec
          +--ro mc-pool

File: YANG/tapi-photonic-media@2018-10-16.tree [hunk 2]
```diff
@@ -183,3 +183,270 @@ module: tapi-photonic-media
...
+          |  +--ro application-code?              string
+          +--ro supportable-modulation*                modulation-technique
+          +--ro total-power-warn-threshold
+             +--ro total-power-upper-warn-threshold-default?   decimal64
+             +--ro total-power-upper-warn-threshold-min?       decimal64
+             +--ro total-power-upper-warn-threshold-max?       decimal64
+             +--ro total-power-lower-warn-threshold-default?   decimal64
+             +--ro total-power-lower-warn-threshold-max?       decimal64
+             +--ro total-power-lower-warn-threshold-min?       decimal64
+  augment /tapi-common:get-service-interface-point-details/tapi-common:output/tapi-common:sip:
```
Comment (arthurMll) [inline YANG/tapi-photonic-media@2018-10-16.tree]: same as above

File: YANG/tapi-photonic-media@2018-10-16.tree (hunk 1 above)
Comment (karthik-sethuraman) [reply YANG/tapi-photonic-media@2018-10-16.tree]: It was already present in the previous version - line 65
[  augment /tapi-common:context/tapi-common:service-interface-point:](https://github.com/OpenNetworkingFoundation/TAPI/blob/0c566fe3b2c8efe925ddd5a7f9948f2df02a9b83/YANG/tapi-photonic-media%402018-10-16.tree#L64)

File: YANG/tapi-photonic-media@2018-10-16.tree (hunk 2 above)
Comment (karthik-sethuraman) [reply YANG/tapi-photonic-media@2018-10-16.tree]: same as above

File: YANG/tapi-photonic-media@2018-10-16.tree (hunk 1 above)
Comment (arthurMll) [reply YANG/tapi-photonic-media@2018-10-16.tree]: Sorry I did not express well the point. My concern is why "get-service-interface-point-list" and "get-service-interface-point-details" are not augmented with:
+--rw media-channel-service-interface-point-spec
       +--ro mc-pool
          +--ro supportable-spectrum* []
          |  +--ro upper-frequency?        uint64
          |  +--ro lower-frequency?        uint64
          |  +--ro frequency-constraint
          |     +--ro adjustment-granularity?   adjustment-granularity
          |     +--ro grid-type?                grid-type
          +--ro available-spectrum* []
          |  +--ro upper-frequency?        uint64
          |  +--ro lower-frequency?        uint64
          |  +--ro frequency-constraint
          |     +--ro adjustment-granularity?   adjustment-granularity
          |     +--ro grid-type?                grid-type
          +--ro occupied-spectrum* []
             +--ro upper-frequency?        uint64
             +--ro lower-frequency?        uint64
             +--ro frequency-constraint
                +--ro adjustment-granularity?   adjustment-granularity
+--ro grid-type? grid-type

the same way the photonic model augment the service interface points data tree here:
[augment /tapi-common:context/tapi-common:service-interface-point:](https://github.com/OpenNetworkingFoundation/TAPI/blob/0c566fe3b2c8efe925ddd5a7f9948f2df02a9b83/YANG/tapi-photonic-media%402018-10-16.tree#L64)

Comment (karthik-sethuraman) [reply YANG/tapi-photonic-media@2018-10-16.tree]: Thanks for the clarification - please review the commit #90bce6a below

**Pull request merged**